src/pudl/analysis/mcoe.py

```diff
@@ -38,7 +39,196 @@
+            "gens": AssetIn(key="denorm_generators_eia"),
+            "frc": AssetIn(key=f"denorm_fuel_receipts_costs_{agg_freqs[freq]}_eia923"),
+        },
+        io_manager_key=io_manager_key,
+        compute_kind="Python",
+    )
+    def fc_asset(
+        hr_by_gen: pd.DataFrame, gens: pd.DataFrame, frc: pd.DataFrame
+    ) -> pd.DataFrame:
+        return fuel_cost(hr_by_gen=hr_by_gen, gens=gens, frc=frc)
+
+    @asset(
+        name=f"capacity_factor_by_generator_{agg_freqs[freq]}",
+        ins={
+            "gen": AssetIn(
+                key=f"generation_fuel_by_generator_{agg_freqs[freq]}_eia923"
+            ),
+            "gens": AssetIn(key="denorm_generators_eia"),
+        },
+        io_manager_key=io_manager_key,
+        compute_kind="Python",
+    )
+    def cf_asset(gens: pd.DataFrame, gen: pd.DataFrame) -> pd.DataFrame:
+        return capacity_factor(gens=gens, gen=gen, freq=freq)
+
+    @asset(
+        name=f"mcoe_{agg_freqs[freq]}",
+        ins={
+            "fuel_cost": AssetIn(key=f"fuel_cost_by_generator_{agg_freqs[freq]}"),
+            "capacity_factor": AssetIn(
+                key=f"capacity_factor_by_generator_{agg_freqs[freq]}"
+            ),
+        },
+        io_manager_key=io_manager_key,
+        compute_kind="Python",
+        config_schema={
+            "min_heat_rate": Field(
+                float,
+                default_value=5.5,
+                description=(
+                    "The lowest plausible heat rate, in mmBTU/MWh. "
+                    "Any MCOE records with lower heat rates are presumed to be "
+                    "invalid, and are discarded before returning."
+                ),
+            ),
+            "min_fuel_cost_per_mwh": Field(
+                float,
+                default_value=0.0,
+                description=(
+                    "The minimum fuel cost on a per MWh basis that is required "
+                    "for a generator record to be considered valid. For some "
+                    "reason there are now a large number of $0 fuel cost records, "
+                    "which previously would have been NaN."
+                ),
+            ),
+            "min_cap_fact": Field(
+                float,
+                default_value=0.0,
+                description=(
+                    "The minimum generator capacity factor. Generator records "
+                    "with a lower capacity factor will be filtered out."
+                    "This allows the user to exclude generators that "
+                    "aren't being used enough to be valid."
+                ),
+            ),
+            "max_cap_fact": Field(
+                float,
+                default_value=1.5,
+                description=(
+                    "The maximum generator capacity factor. Generator records "
+                    "with a higher capacity factor will be filtered out."
+                ),
```

Comment:
I added a context input to parameterize the bounds for these values. The default value that was in there for `max_cap_fact` was 1.5. Why not 1.0? Should I leave as 1.5?